src/commands.rs

```diff
@@ -302,3 +345,54 @@ impl CommandErrorHandler for CloseSocketCommand {
         StackError::CloseError(error)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use embedded_nal::{Ipv4Addr, Ipv6Addr};
+    use heapless::String;
+
+    use super::MAX_IP_LENGTH;
+
+    macro_rules! test_v4 {
```

Comment:
Elegant solution. Just out of curiosity, is there an advantage to using macros here instead of a helper method?

Reply:
I think a helper function would show up in the error trace (so the assertion would fail in `test_v4` and not in `test_ipv4_to_string`). However, I didn't actually test it, to be honest, and a helper function might work as well 🙂